lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1565,22 +1565,22 @@ SPIRVToLLVMDbgTran::transDebugIntrinsic(const SPIRVExtInst *DebugInst,
       // parameter. To work around this limitation we create a dummy temp
       // alloca, use it to create llvm.dbg.declare, and then remove the alloca.
       auto *AI = new AllocaInst(Type::getInt8Ty(M->getContext()), 0, "tmp", BB);
-      auto *DbgDeclare = getDIBuilder(DebugInst).insertDeclare(
+      DbgInstPtr DbgDeclare = getDIBuilder(DebugInst).insertDeclare(
           AI, LocalVar.first, GetExpression(Ops[ExpressionIdx]),
           LocalVar.second, BB);
       AI->eraseFromParent();
-      return DbgDeclare;
+      return cast<Instruction *>(DbgDeclare);
```

Comment:
Thanks! 

Maybe we can use `.get<Instruction *>()` instead of cast?

[llvm/lib/Transforms/Utils/Local.cpp](https://github.com/llvm/llvm-project/commit/9997e0397156#diff-260cb3f00c79bee922ae9d3c19da751208997ad5c919d5cdf412c2259e0445eb)

Reply:
Sure that seems to work too; I've updated the patch.